From e3cf18baf5412966434e31d90f4ee7325f617ab1 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 19 Jul 2018 10:53:40 +0900
Subject: [PATCH 331/909] drm: bridge/dw-hdmi: Fix offset address of register
 for R-Car

commit 69a962e8a8be8c31dd3b8fbd7ae9e42f5105370b from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

By H/W manual update, fc_vsdieeeid0 register has been corrected to 0x1031
and fc_vsdieeeid2 register has been corrected to 0x1029.
So this patch fixes its offset address for R-Car.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 12 +++++++++---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.h |  2 ++
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 16c45305fe7f..a29c0bcbf60a 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -1533,9 +1533,15 @@ static void hdmi_config_vendor_specific_infoframe(struct dw_hdmi *hdmi,
 	hdmi_writeb(hdmi, buffer[2], HDMI_FC_VSDSIZE);
 
 	/* Set 24bit IEEE Registration Identifier */
-	hdmi_writeb(hdmi, buffer[4], HDMI_FC_VSDIEEEID0);
-	hdmi_writeb(hdmi, buffer[5], HDMI_FC_VSDIEEEID1);
-	hdmi_writeb(hdmi, buffer[6], HDMI_FC_VSDIEEEID2);
+	if (hdmi->plat_data->dev_type == RCAR_HDMI) {
+		hdmi_writeb(hdmi, buffer[4], HDMI_RCAR_FC_VSDIEEEID0);
+		hdmi_writeb(hdmi, buffer[5], HDMI_FC_VSDIEEEID1);
+		hdmi_writeb(hdmi, buffer[6], HDMI_RCAR_FC_VSDIEEEID2);
+	} else {
+		hdmi_writeb(hdmi, buffer[4], HDMI_FC_VSDIEEEID0);
+		hdmi_writeb(hdmi, buffer[5], HDMI_FC_VSDIEEEID1);
+		hdmi_writeb(hdmi, buffer[6], HDMI_FC_VSDIEEEID2);
+	}
 
 	/* Set HDMI_Video_Format and HDMI_VIC/3D_Structure */
 	hdmi_writeb(hdmi, buffer[7], HDMI_FC_VSDPAYLOAD0);
diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.h b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.h
index 3d00338499f0..c7ea6e2d3352 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.h
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.h
@@ -109,9 +109,11 @@
 #define HDMI_FC_AUDICONF2                       0x1027
 #define HDMI_FC_AUDICONF3                       0x1028
 #define HDMI_FC_VSDIEEEID0                      0x1029
+#define HDMI_RCAR_FC_VSDIEEEID2                 0x1029
 #define HDMI_FC_VSDSIZE                         0x102A
 #define HDMI_FC_VSDIEEEID1                      0x1030
 #define HDMI_FC_VSDIEEEID2                      0x1031
+#define HDMI_RCAR_FC_VSDIEEEID0                 0x1031
 #define HDMI_FC_VSDPAYLOAD0                     0x1032
 #define HDMI_FC_VSDPAYLOAD1                     0x1033
 #define HDMI_FC_VSDPAYLOAD2                     0x1034
-- 
2.17.1

